app/main/appium-method-handler.js

```diff
@@ -167,46 +169,120 @@ export default class AppiumMethodHandler {
+    if (currentContext !== NATIVE_APP) {
+      await this.driver.context(NATIVE_APP);
     }
 
+    ({platformName, statBarHeight} = await this.driver.sessionCapabilities());
+
     try {
-      source = await this.driver.source();
+      windowSize = await this.driver.getWindowSize();
     } catch (e) {
       if (e.status === 6) {
         throw e;
       }
-      sourceError = e;
+      windowSizeError = e;
     }
 
     try {
-      screenshot = await this.driver.takeScreenshot();
+      contexts = await this._getContexts(platformName);
     } catch (e) {
       if (e.status === 6) {
         throw e;
       }
-      screenshotError = e;
+      contextsError = e;
     }
 
+    if (currentContext !== NATIVE_APP) {
+      await this.driver.context(currentContext);
+    }
+    // End of note
+
+    /**
+     * If its a webview then update the HTML with the element location
+     * so the source can be used in the native inspector
+     */
     try {
-      windowSize = await this.driver.getWindowSize();
+      if (currentContext !== NATIVE_APP) {
+        const webviewStatusAddressBarHeight = await this.driver.execute(getWebviewStatusAddressBarHeight, [{platformName, statBarHeight}]);
 
+        await this.driver.execute(setHtmlElementAttributes, [{platformName, webviewStatusAddressBarHeight}]);
+      }
+    } catch (ign) {}
+
+    try {
+      source = parseSource(await this.driver.source());
     } catch (e) {
       if (e.status === 6) {
         throw e;
       }
-      windowSizeError = e;
+      sourceError = e;
+    }
+
+    try {
+      screenshot = await this.driver.takeScreenshot();
+    } catch (e) {
+      if (e.status === 6) {
+        throw e;
+      }
+      screenshotError = e;
     }
 
-    return {contexts, contextsError, source, sourceError, screenshot, screenshotError, windowSize, windowSizeError};
+    return {contexts, contextsError, currentContext, currentContextError,
+            source, sourceError, screenshot, screenshotError, windowSize, windowSizeError};
+  }
+
+  /**
+   * Retrieve available contexts, along with the title associated with each webview
+   */
+  async _getContexts (platform) {
+    return platform.toLowerCase() === 'ios' ? await this.driver.execute('mobile:getContexts', []) : await this._getAndroidContexts();
+  }
+
+  /**
+   * Custom implementation of the `mobile:getContexts` for Android, which only returns an object with the id and title
+   */
+  async _getAndroidContexts () {
+    let newContexts = [];
+    const currentContext = await this.driver.currentContext();
+
+    // Get the contexts and retrieve extra data
+    const contexts = await this.driver.contexts();
+    const getContextData = async (context) => {
+      let title;
+      const id = context;
+      if (id !== NATIVE_APP) {
+        await this.driver.context(context);
+        const pageTitle = await this.driver.title();
+        title = pageTitle === '' ? 'No page title available' : pageTitle;
```

Comment:
`pageTitle === ''` could be `_.isEmpty(pageTitle)`  (lodash)

Reply:
Fixed it, thanks